lib/src/components/NxThreatIndicatorLegend/NxThreatIndicatorLegend.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef } from 'react';
+import classnames from 'classnames';
+import {
+  NxThreatIndicatorLegendProps,
+  nxThreatIndicatorLegendPropTypes,
+  ThreatProps
+} from './types';
+import NxThreatIndicator from '../NxThreatIndicator/NxThreatIndicator';
+import './NxThreatIndicatorLegend.scss';
+
+const capitalize = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
+const ThreatIndicator = ({ threatType }: ThreatProps) => {
+  return (
+    <div className="nx-threat-container">
+      <NxThreatIndicator threatLevelCategory={threatType} />
+      <span>{capitalize(threatType)}</span>
+    </div>
+  );
+};
+
+const NxThreatIndicatorLegend = forwardRef<HTMLDivElement, NxThreatIndicatorLegendProps>(
+    (props: NxThreatIndicatorLegendProps, ref) => {
+      const {
+        critical,
+        severe,
+        moderate,
+        low,
+        none,
+        unspecified,
+        vertical,
+        header,
+        className,
+        ...attrs
+      } = props;
+      const classNames = classnames(className, 'nx-threat-indicator-legend', {
+        'nx-threat-indicator-legend--vertical': vertical
+      });
+
+      if (!critical && !severe && !moderate && !low && !none && !unspecified) {
+        console.warn(`No threat level categories for NxThreatIndicatorLegend have been provided
+        as props so nothing will be rendered.`);
+        return null;
+      }
+
+      return (
+        <div ref={ref} className={classNames} {...attrs}>
+          <label className="nx-threat-indicator-legend-header">{header || 'Legend'}</label>
```

Comment:
This class should probably be `nx-threat-indicator-legend`

Reply:
> This class should probably be `nx-threat-indicator-legend`

I'm a bit confused. The `nx-threat-indicator-legend` class is already applied on the main `div`. The `nx-threat-indicator-legend-header` is applied to the `label` as the legend header formatting. Could you please clarify?

Reply:
> I'm a bit confused. The nx-threat-indicator-legend class is already applied on the main div. The nx-threat-indicator-legend-header is applied to the label as the legend header formatting. Could you please clarify?

Sorry, I must have lost track of what I was saying partway through.  I meant it should be `nx-threat-indicator-legend__header`

Reply:
> > I'm a bit confused. The nx-threat-indicator-legend class is already applied on the main div. The nx-threat-indicator-legend-header is applied to the label as the legend header formatting. Could you please clarify?
> 
> Sorry, I must have lost track of what I was saying partway through. I meant it should be `nx-threat-indicator-legend__header`

No worries! It's changed here:
1a4c6bee156c3bb6149a6ab31217e58137ad8b1c